src/Adapter/Image/Uploader/AbstractImageUploader.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Image\Uploader;
+
+use ImageManager;
+use PrestaShop\PrestaShop\Core\Image\Uploader\Exception\UploadedImageConstraintException;
+use PrestaShop\PrestaShop\Core\Image\Uploader\ImageUploaderInterface;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Tools;
+
+/**
+ * Class AbstractImageUploader encapsulates reusable legacy methods used for image uploading.
+ *
+ * @internal
+ */
+abstract class AbstractImageUploader implements ImageUploaderInterface
+{
+    /**
+     * Check if image is allowed to be uploaded.
+     *
+     * @param UploadedFile $image
+     *
+     * @throws UploadedImageConstraintException
+     */
+    protected function checkImageIsAllowedForUpload(UploadedFile $image)
+    {
+        $maxFileSize = Tools::getMaxUploadSize();
+
+        if ($maxFileSize > 0 && $image->getSize() > $maxFileSize) {
+            throw new UploadedImageConstraintException(
+                sprintf(
+                    'Max file size allowed is "%s" bytes. Uploaded image size is "%s".',
+                    $maxFileSize,
+                    $image->getSize()
+                ),
+                UploadedImageConstraintException::EXCEEDED_SIZE
+            );
+        }
+
+        if (!ImageManager::isRealImage($image->getPathname(), $image->getClientMimeType())
+            || !ImageManager::isCorrectImageFileExt($image->getClientOriginalName())
+            || preg_match('/\%00/', $image->getClientOriginalName())
```

Comment:
:+1: Maybe add a comment to says it to prevent the null byte injection (even if this code comes from legacy)

Reply:
i didnt know that to be honest.. ^^

Reply:
Little explanation: http://www.madirish.net/401

Reply:
thanks. :)